Add first Gremlin

Xiphe opened the pull request:
<!-- decorate-gh-pr -->
<a href="https://teamsgame.agilepainrelief.com/preview/gremlin-team-management-rel"><img src="https://img.shields.io/badge/published-gh--pages-green" alt="published to gh-pages" /></a><hr />
<!-- /decorate-gh-pr -->

This PR introduces the first Gremlin "Team Member/Management Relationship"

Xiphe commented on src/state/gremlins.spec.ts:
```diff
@@ -0,0 +1,65 @@
+    });
+  };
+
+  describe('Team Member/Management Relationship', () => {
```
@mlevison could you please review if the acceptance of the tests are correct?

Xiphe commented on src/state/gremlins.spec.ts:
```diff
@@ -0,0 +1,65 @@
+    });
+  };
+
+  describe('Team Member/Management Relationship', () => {
```
@mlevison could you please review if the acceptance of the tests are correct?

Xiphe replied:
not sure what "until something is done" means in game logic.

mlevison commented:
It's complicated 
- First that gremlin was poorly designed and needs to replaced, something like this is so rare as to be not worth using as an example.
- Implementing a the working agreements or something else that improves the team communications/social interactions merely stops the bleeding it doesn't magically restore things. _Consider if something like this happened in real life. The damage is done in the moment, attempts to repair it later don't fix the damage_

Separate from this:
- Testing philosophy - I would consider creating a fake social/communication action - I would use that in my test cases when we don't care which action is played. When we test with an action like the "working agreements" we must keep track of the effect on capacity as we read the tests. If we use a fake action with no effect then we only pay attention to see if halted the gremlin.

(as usual a simple thing and I write too many words)
Mark

Xiphe commented:
OK so what should I do?

- Not implement this gremlin at all?
- Implement another one?
- Implement this one simpler? (How?)

---

For Testing I agree that some action stubs that do nothing but effecting the currently tested action makes sense.

mlevison commented:
Implement this gremlin: 


Event | Effect | Die   Roll | Mitigating   Factors known   only to the facilitator
-- | -- | -- | --
We’ve had an   emergency on another team, we need your best tester for a while. | Capacity - 3 for   three rounds | 4 | Teams whose ScrumMaster   has been protecting them from distractions reduce the number of rounds by 1.       Informal or   Formal cross-training also reduces the impact by 1 Capacity Story Point.       These are cumulative   effects.

Instead of die roll lets just give it a 10% chance of happening each round. Perhaps we also place a limit of one gremlin per round. And no gremlin in the first two rounds.

Xiphe commented:
I'll pause here and continue tomorrow.

Status:
- added `getGame()` test helper to reduce boilerplate in tests
- added [specs for "emergency on another team"](https://github.com/mlevison/high-performance-teams-game/pull/12/files#diff-4481d05987dd5dd952bbe179e9d4d6a35432b285919b96151216988024418df0)
- prepared for mocking out gameAction costs and effects. (Instead of using Fake-Game actions I'd strap of effects and costs from existing ones. That way we can check for specific `gameActionId`s to determine the effect instead of introducing other identifying factors like `type`.

Next: Actually use gremlins in `useAppState()` and implement the first gremlin.

Xiphe commented:
Implementation is done.

The getGremlin logic is currently uncovered.
